src/esp/assets/GenericMeshData.cpp

```diff
@@ -72,15 +70,23 @@ void GenericMeshData::setMeshData(Magnum::Trade::AbstractImporter& importer,
     collisionMeshData_.indices = meshData_->mutableIndices<Mn::UnsignedInt>();
   else
     collisionMeshData_.indices = indexData_ = meshData_->indicesAsArray();
-}
+}  // setMeshData
 
-void GenericMeshData::setMeshData(Magnum::Trade::AbstractImporter& importer,
-                                  const std::string& meshName) {
-  // make sure name is appropriate for importer
-  int meshID = importer.meshForName(meshName);
-  CORRADE_ASSERT(meshID != -1, "Unknown meshName :" << meshName, );
-  setMeshData(importer, meshID);
-}
+void GenericMeshData::importAndSetMeshData(
+    Magnum::Trade::AbstractImporter& importer,
+    int meshID) {
+  /* Guarantee mesh instance success */
+  CORRADE_INTERNAL_ASSERT_OUTPUT(meshData_ = importer.mesh(meshID));
+  setMeshData(meshData_);
```

Comment:
Not sure about this. It sets the `meshData_` member and then it calls `setMeshData()` which takes a reference to `meshData_` and sets it to `meshData_` again. If there wouldn't be the `interleave()` in between, that would basically be a self-assignment ;) What about this instead (together with the `setMeshData()` signature change mentioned above)?

```suggestion
  Cr::Containers::Optional<Mn::Trade::MeshData> mesh = importer.mesh(meshID));
  CORRADE_INTERNAL_ASSERT(mesh);
  setMeshData(*std::move(mesh));
```

Same in the function below.

Reply:
I was thinking about that, but I figured it was nice to have things in place. -shrug-. 

how about this, so that the import is enclosed (or is this not necessary i.e. will importer never fail?):

`  Cr::Containers::Optional<Mn::Trade::MeshData> mesh;
  CORRADE_INTERNAL_ASSERT_OUTPUT(mesh = importer.mesh(meshID));
`

Reply:
Um... not sure about your point, your code and the first two lines in mine do practically the same, it's just that the assignment happens later.

Reply:
Wasn't sure if we wanted to wrap the access/assignment in the assertion.  Guess that's not necessary.

Reply:
Ultimately the assertion needs to check that `mesh` is not `NullOpt`, and it doesn't matter if it's done on the result of the assignment expression or separately. Apart from `mesh` being a null optional there's no other case where something could fail.